File: apps/api/schemas/job.py
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


class JobSpec(BaseModel):
    name: str
    owner: str
    team: str
    priority: Literal["low", "normal", "high"] = "normal"
    resources: dict[str, Any] | None = None
    limits: dict[str, Any] | None = None
    env: dict[str, str] | None = None
    runtime: dict[str, Any]
    artifacts: list[str] | None = None
    tags: list[str] | None = None


class JobCreate(BaseModel):
    yaml: str | None = None
    spec: JobSpec | None = None


class JobOut(BaseModel):
    id: str
    name: str
    team_id: str
    owner_id: str
    priority: str
    state: str
    queued_at: datetime | None = None
    started_at: datetime | None = None
    finished_at: datetime | None = None

    class Config:
        from_attributes = True


class JobListOut(BaseModel):
    items: list[JobOut]
    total: int
    page: int
    limit: int

File: apps/web/src/schemas/job.ts
import { z } from "zod";

export const JobOut = z.object({
  id: z.string(),
  name: z.string(),
  team_id: z.string(),
  owner_id: z.string(),
  priority: z.string(),
  state: z.string(),
  queued_at: z.string().nullable(),
  started_at: z.string().nullable(),
  finished_at: z.string().nullable(),
});

export const JobListOut = z.object({
  items: z.array(JobOut),
  total: z.number(),
  page: z.number(),
  limit: z.number(),
});

export type Job = z.infer<typeof JobOut>;
export type JobList = z.infer<typeof JobListOut>;
